### ISSUE TITLE
URL Callback

### ISSUE THREAD

**Issue by danieldocki** [mcorp/locasms#16]
Place the callback URL would be great. It intends to do this?

```
http://54.173.24.177/painel/api.ashx?action=sendsms&lgn=&pwd=&msg=&numbers=319413 6747&url_callback=<URL>
```

**Comment by vyper** [comment]
@danieldocki can you are open a PR with this feature? (:

**Comment by danieldocki** [comment]
@vyper It does so is good?

```
def deliver(message, *mobiles, **opts)
  attrs = {
    msg: message,
    numbers: numbers(mobiles)
  }.merge(opts)
  rest.get(:sendsms, attrs)['data']
end


# delivering message to one mobile
cli.deliver 'my message', '1155559999'

# delivering message to one mobile with callback
cli.deliver 'my message', '1155559999', url_callback: 'http://mysite.com/callback'

# delivering the same message to multiple mobliles at once
cli.deliver 'my message', '1199998888,5500002222'
cli.deliver 'my message', '1199998888', '5500002222'
cli.deliver 'my message', ['1199998888', '5500002222']
cli.deliver 'my message', %w(1199998888 5500002222)

# delivering the same message to multiple mobliles at once with callback
cli.deliver 'my message', '1199998888,5500002222', url_callback: 'http://mysite.com/callback'
cli.deliver 'my message', '1199998888', '5500002222', url_callback: 'http://mysite.com/callback'
cli.deliver 'my message', ['1199998888', '5500002222'], url_callback: 'http://mysite.com/callback'
cli.deliver 'my message', %w(1199998888 5500002222), url_callback: 'http://mysite.com/callback'
```

```
def deliver_at(message, datetime, *mobiles, **opts)
  date, time = Helpers::DateTimeHelper.split datetime
  attrs = {
    msg: message,
    numbers: numbers(mobiles),
    jobdate: date,
    jobtime: time
  }.merge(opts)
  rest.get(:sendsms, attrs)['data']
end

# scheduling the deliver of a message to one mobile
cli.deliver_at 'my message', '2013-10-12 20:33:00', '1155559999'

# scheduling the deliver of a message to one mobile with callback
cli.deliver_at 'my message', '2013-10-12 20:33:00', '1155559999', url_callback: 'http://mysite.com/callback'

# scheduling the deliver of a message to multiple mobiles at once
cli.deliver_at 'my message', '2013-10-12 20:33:00', '1199998888,5500002222'
cli.deliver_at 'my message', '2013-10-12 20:33:00', '1199998888', '5500002222'
cli.deliver_at 'my message', '2013-10-12 20:33:00', ['1199998888', '5500002222']
cli.deliver_at 'my message', '2013-10-12 20:33:00', %w(1199998888 5500002222)

# scheduling the deliver of a message to multiple mobiles at once with callback
cli.deliver_at 'my message', '2013-10-12 20:33:00', '1199998888,5500002222', url_callback: 'http://mysite.com/callback'
cli.deliver_at 'my message', '2013-10-12 20:33:00', '1199998888', '5500002222', url_callback: 'http://mysite.com/callback'
cli.deliver_at 'my message', '2013-10-12 20:33:00', ['1199998888', '5500002222'], url_callback: 'http://mysite.com/callback'
cli.deliver_at 'my message', '2013-10-12 20:33:00', %w(1199998888 5500002222), url_callback: 'http://mysite.com/callback'
```

**Comment by adilsoncarvalho** [comment]
If the callback is the same to all requests doesn't make sense to keep it as a configuration?

**Comment by vyper** [comment]
@adilsoncarvalho in [locasms doc](http://locasms.com.br/download/locasms-manual-api.pdf) specified the `url_callback` in each call of the sendsms.

**Comment by vyper** [comment]
@danieldocki for me it's so ugly, but... locasms works well 😁

:+1:

**Comment by adilsoncarvalho** [comment]
I see but we could have is a configuration option that would provide a default callback that could be override on a call. What do you think about that @danieldocki ?

**Comment by danieldocki** [comment]
@adilsoncarvalho I think it's okay. How can we design it?

**Comment by tiagopog** [comment]
It's interesting to think that there can be cases where clients of this gem are using it to serve SMS' for many applications with distinct domains. Probably a good design would provide the "url_callback" key argument with a fallback to a default URL.

**Comment by danieldocki** [comment]
@tiagopog I agree with you 👍

**Comment by AlessandroMinali** [comment]
Happy Hacktoberfest! I'll be looking into this!

**Comment by adilsoncarvalho** [comment]
Solved on #18
